airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py

```diff
@@ -447,6 +447,39 @@ class TestStreamConnection(GoogleAnalyticsV4Stream):
     Because of the nature of the connector, the streams are created dynamicaly.
     We declare the static stream like this to be able to test out the prmissions to read the particular view_id."""
 
+    page_size = 1
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """For test reading pagination is not required"""
+        return None
+
+    def request_body_json(
+        self, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> Optional[Mapping]:
+        """
+        Overwrite the method to fetch records from start_date up to now for testing case,
+        and request 1 record per page
+        """
+        metrics = [{"expression": metric} for metric in self.metrics]
+        dimensions = [{"name": dimension} for dimension in self.dimensions]
+
+        start_date = pendulum.parse(self.start_date).date()
```

Comment:
how about instead of overriding this method you override stream_slices to return [start_date, end_date] ?

Reply:
@keu Done.